pkg/stack/unwind/unwind_table.go

```diff
@@ -261,9 +265,11 @@ func (ptb *UnwindTableBuilder) readFDEs(path string) (frame.FrameDescriptionEntr
 }
 
 func buildUnwindTable(fdes frame.FrameDescriptionEntries) UnwindTable {
-	table := make(UnwindTable, 0, len(fdes))
+	table := make(UnwindTable, 0)
 	for _, fde := range fdes {
-		table = append(table, buildTableRows(fde)...)
+		forEachUnwindRow(fde, func(unwindRow UnwindTableRow) {
```

Comment:
I'm sorry, but I need to reject this hard. I don't understand what this provides over simply returning table rows and consuming those rows in different places. There is no redundancy saving or similar. This makes this code more complicated to read. Not to mention this is not idiomatic Go. Also, `callback` implies that this is something async, but it is not.

Reply:
You are correct that in this function there are no savings. We are still allocating the slice without even doing preallocation right. This is obviously not great, and `buildUnwindTable` does a lot of allocations.

Sorry that I wasn't super clear in the description. This was done on purpose as I didn't want to change how the table is generated for BPF. This PR is just about the parser and how we moved the responsibility of allocation of the table from the parser to the caller. 

I would prefer to produce the full table for this particular use case in this PR. The plan was to fully remove `buildUnwindTable` in a forthcoming PR that changes how the table is generated (https://github.com/parca-dev/parca-agent/issues/1052).

Reply:
We discussed the issue offline. And we are trying to find a compromise here. If we can't, we agree to keep the pattern because the gains are too impressive to neglect.